include/triton/core/tritonserver.h

```diff
@@ -1071,6 +1071,20 @@ TRITONSERVER_DECLSPEC TRITONSERVER_Error* TRITONSERVER_InferenceRequestAddInput(
     const TRITONSERVER_DataType datatype, const int64_t* shape,
     uint64_t dim_count);
 
+/// Add a raw input to a request. The model input name, data type and shape of
+/// the input will be deduced from model configuration, this function must at
+/// most be called once on request with no other input to ensure the deduction
```

Comment:
".This function must be called at most once per request for a single input. This ensures the deduction will be accurate."